### PR TITLE
Reverse order of CRD init operations

### PR DESCRIPTION
**Pull request by schancel**
This commit reverses the order of CRD init operations. The reason for
doing CRD init operations in this new order, is so that k8s admins
can manually create the expected CRDs. By creating the CRDs by hand,
create CRD permission does not need to be granted to the nats-operator SA.
Only read permission is required.

### CODE REVIEW

File: pkg/util/kubernetes/crd.go [hunk 1]
```diff
@@ -128,35 +128,28 @@ func WaitCRDs(extsClient extsclientset.Interface) error {
...
 	}
-	if !IsKubernetesResourceAlreadyExistError(err) {
-		// The crd doesn't exist yet but we've got an unexpected error while registering it.
-		return err
+
+	// CRD already exists, and is different than what is expected.
+	if err == nil {
+		// Attempt to update the CRD by setting its spec to the expected value.
+		d.Spec = crd.Spec
+		return extsClient.ApiextensionsV1beta1().CustomResourceDefinitions().Update(d)
```
Comment (wallyqs) [inline pkg/util/kubernetes/crd.go]:
```suggestion
		_, err = extsClient.ApiextensionsV1beta1().CustomResourceDefinitions().Update(d)
		return err
```

File: pkg/util/kubernetes/crd.go [hunk 2]
```diff
@@ -128,35 +128,28 @@ func WaitCRDs(extsClient extsclientset.Interface) error {
...
 	}
 
-	// At this point the CRD already exists but its spec may differ.
-	// This can happen, for instance, if the user has tampered with the CRD.
-	// Hence, we do our best to bring in line with our expectations.
-	// Fetch the latest version of the CRD.
-	d, err := extsClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(crd.Name, metav1.GetOptions{})
+	// No CRD existed, attempt to register the CRD.
+	_, err := extsClient.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crd)
```
Comment (wallyqs) [inline pkg/util/kubernetes/crd.go]:
```suggestion
	_, err = extsClient.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crd)
```

**Comment by wallyqs** [comment]
Thanks @schancel for the PR! Left a couple of suggestions that I think will fix the error in the build.

**Comment by schancel** [comment]
@wallyqs Should be fixed.  Any chance you're willing to switch to `go mod`?  It would make it easier to build.  I can't actually build this right now, without changing my go setup.

File: pkg/util/kubernetes/crd.go [hunk 3]
```diff
@@ -128,35 +128,29 @@ func WaitCRDs(extsClient extsclientset.Interface) error {
...
-		return err
-	}
-	if reflect.DeepEqual(d.Spec, crd.Spec) {
-		// The specs match, so there's nothing to do.
-		return nil
-	}
-	// Attempt to update the CRD by setting its spec to the expected value.
-	d.Spec = crd.Spec
-	if _, err := extsClient.ApiextensionsV1beta1().CustomResourceDefinitions().Update(d); err != nil {
+		// Registration failed
```
Comment (schancel) [inline pkg/util/kubernetes/crd.go]: Actually this is silly...

**Comment by wallyqs** [comment]
Sorry about the inconvenience, will switch to use go modules after this.

**Pull request merged**